### PR TITLE
update package to 1.10

### PR DESCRIPTION
**Pull request by JamesRobertsonGames**
lazy-object-proxy 1.10 

**Destination channel:** defaults

### Links

- [PKG-4434](https://anaconda.atlassian.net/browse/PKG-4434) 
- [Upstream repository](https://github.com/ionelmc/python-lazy-object-proxy)
- [Upstream changelog/diff](https://github.com/ionelmc/python-lazy-object-proxy/releases/tag/v1.10.0)
- Relevant dependency PRs:
  - moto

### Explanation of changes:
- update version and update testing

[PKG-4434]: https://anaconda.atlassian.net/browse/PKG-4434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,40 +1,49 @@
...
 requirements:
   build:
     - python                                 # [build_platform != target_platform]
     - {{ compiler('c') }}
   host:
     - python
+    - wheel
     - pip
-    - setuptools_scm >=3.3.1
+    - setuptools_scm >=8.0.0
```
Comment (cbouss) [inline recipe/meta.yaml:25]:
```suggestion
    - setuptools >=64
    - setuptools_scm >=8.0.0
```
https://github.com/ionelmc/python-lazy-object-proxy/blob/v1.10.0/pyproject.toml#L3

Comment (skupr-anaconda) [inline recipe/meta.yaml:25]:
```suggestion
    - setuptools
    - setuptools_scm >=8.0.0
```
Missing setuptools, see https://github.com/ionelmc/python-lazy-object-proxy/blob/aa46e846fa2eb7cb845a07e122f2984267551098/pyproject.toml#L3

**Review by skupr-anaconda — APPROVED** [review]
LGTM but `setuptools`

**Review by Jrice1317 — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,40 +1,50 @@
...
 requirements:
   build:
     - python                                 # [build_platform != target_platform]
     - {{ compiler('c') }}
   host:
     - python
+    - wheel
     - pip
-    - setuptools_scm >=3.3.1
+    - setuptools
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - setuptools >=64
```

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**